deeposlandia/network.py

```diff
@@ -137,17 +155,35 @@ def dense(self, x, depth, dropout_rate=1.0, activation='relu', batch_norm=True):
         batch_norm : boolean
             If True, a batch normalization process is applied on `x` tensor before activation
             layer
+        name : str
+            Fully-connected block name, for identification purpose
 
         Returns
         -------
         tensor
-            2D output layer
+            Output layer
         """
-        x = K.layers.Flatten()(x)
-        x = K.layers.Dense(depth)(x)
+        x = K.layers.Dense(depth, name=name+'_fc')(x)
         if batch_norm:
-            x = K.layers.BatchNormalization()(x)
-        x = K.layers.Activation(activation)(x)
-        x = K.layers.Dropout(dropout_rate)(x)
+            x = K.layers.BatchNormalization(name=name+'_bn')(x)
+        x = K.layers.Activation(activation, name=name+'_activation')(x)
+        x = K.layers.Dropout(dropout_rate, name=name+'_dropout')(x)
         return x
 
+    def flatten(self, x, name=""):
```

Comment:
Same comment about the `name` parameter.

Reply:
Seen.